statsmodels/sandbox/regression/gmm.py

```diff
@@ -70,38 +73,36 @@ def maxabs(x):
 
 
 class IV2SLS(LikelihoodModel):
-    '''
-    Class for instrumental variables estimation using Two-Stage Least-Squares
+    """
+    Instrumental variables estimation using Two-Stage Least-Squares (2SLS)
 
 
     Parameters
     ----------
-    endog: array 1d
-       endogenous variable
+    endog: array
+       Endogenous variable, 1-dimensional or 2-dimensional array nobs by 1
     exog : array
-       explanatory variables
+       Explanatory variables, 1-dimensional or 2-dimensional array nobs by k
     instruments : array
-       instruments for explanatory variables, needs to contain those exog
-       variables that are not instrumented out
+       Instruments for explanatory variables. Must contain both exog
+       variables that are not being instrumented and instruments
 
     Notes
     -----
     All variables in exog are instrumented in the calculations. If variables
-    in exog are not supposed to be instrumented out, then these variables
-    need also to be included in the instrument array.
+    in exog are not supposed to be instrumented, then these variables
+    must also to be included in the instrument array.
 
     Degrees of freedom in the calculation of the standard errors uses
     `df_resid = (nobs - k_vars)`.
     (This corresponds to the `small` option in Stata's ivreg2.)
-
-
-    '''
+    """
 
     def __init__(self, endog, exog, instrument=None):
-        self.instrument = instrument
+        self.instrument = _ensure_2d(instrument)
```

Comment:
If it works fine, then it's by chance. pandas/numpy compatibility has improved over time, but there are still many differences, that could cause subtle bugs.
endog and exog are converted to ndarrays in super.